staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -404,6 +404,121 @@ func TestGetServerResourcesForGroupVersion(t *testing.T) {
 	}
 }
 
+func TestHashes(t *testing.T) {
+	savedHashes := map[string][]string{}
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		savedHashes[req.URL.Path] = req.URL.Query()["hash"]
+		var obj interface{}
+		switch req.URL.Path {
+		case "/api":
+			obj = &metav1.APIVersions{
+				Versions: []string{
+					"v1",
+				},
+			}
+		case "/apis":
+			obj = &metav1.APIGroupList{
+				Groups: []metav1.APIGroup{
+					{
+						Name: "extensions",
+						Versions: []metav1.GroupVersionForDiscovery{
+							{GroupVersion: "extensions/v1beta1", Version: "v1beta1"},
+							{GroupVersion: "extensions/v1beta2", Version: "v1beta2"},
+						},
+						Hash: "hash1",
```

Comment:
Yeah, that's almost certainly wrong, if you look at the fold hierarchy in your `.kube/cache/discovery`, if you find a group that has multiple version, each version has a different file, which means the two files have different hashes, so we need one hash per GroupVersion.

Reply:
done, see above